lib/spack/spack/cmd/testsuite.py

```diff
@@ -0,0 +1,221 @@
+    
+    for compilers in spack.compilers.all_compilers(scope=None):
+        systemCompilers.append(str(compilers))
+
+    for compiler in systemCompilers:
+        if compiler in newMaster:
+            newList.append(compiler)
+    return newList
+
+
+'''
+fucntion Name: removeTests
+parameters:
+    testList  : list
+    exclusion : str
+returns:
+    list
+
+Receieves a list of all tests and an excusion.
+Depending on the type of exclusion it will remove tests matching.
+Returns the list with the tests matching the excusion removed.
+
+exclusions can be in this form:
+pkg%compiler
+pkg@version
+compiler@version
+pkg@version%compiler@version
+pkg@version%compiler
+pkg%compiler@version
+pkg
+compiler
+'''
+def removeTests(testList, exclusion):
+    pkg = ""
+    compiler=""
+    teststoRemove = []
+    if "%" in exclusion:
+        pkg = exclusion.split('%')[0]
+        compiler=exclusion.split('%')[1]
+        if '@' in pkg and '@' in compiler:#pkg@version%compiler@version
+            if exclusion in testList:
+                teststoRemove.append(exclusion)
+        elif '@' in pkg or '@' in compiler:#pkg@version%compiler or pkg%compiler@version
+            for test in testList:
+                if pkg in test and compiler in test:
+                    teststoRemove.append(test)
+        else:#pkg%compiler
+            for test in testList:
+                if pkg in test and compiler in test:
+                    teststoRemove.append(test)
+    elif '@' in exclusion: #pkg@version or compiler@version
+        for test in testList:
+            if exclusion in test:
+                teststoRemove.append(test)
+    else:#pkg or compiler
+        remove=0
+        for test in testList:
+            if exclusion in str(test):
+                teststoRemove.append(test)
+    for test in teststoRemove: #remove tests from main test list
+        testList.remove(test)
+    return testList
+
+def testsuite(parser, args):
+    if not args.yamlFile:
+        tty.die("spack testsuite requires a yaml file as argument.")
+    if args.complete:
+        args.log_format='cdash-complete' #use cdash-complete if you want configure, build and test output.
+        cdash = '--log-format=cdash-complete'
+    else:
+        args.log_format='cdash-simple' #use cdash-complete if you want configure, build and test output.
+        cdash = '--log-format=cdash-simple'
+    print args.complete
+    enabledTests = []
+    compilers = []
+    packages = []
+    tests = []
+    testPackages = []
+    versions = []
+    packageVersion = []
+    compilerVersion = []
+    path = ""
+    dashboards = []
+    #designed to use a single file and modify the enabled tests, thus requiring a single file modification.
+    for files in args.yamlFile: #read yaml files which contains description of tests
+        with open(files, 'r') as stream:
+            try:
+                yamlFile= yaml.load(stream)
+            except yaml.YAMLError as exc:
```

Comment:
Wrap this up in a `SpackYAMLError` so that the error will be caught and turned into a nice error message on failure. See [Spec.from_yaml](https://github.com/LLNL/spack/blob/develop/lib/spack/spack/spec.py#L1319) for an example.